src/Http/Client.php

```diff
@@ -549,3 +549,5 @@ protected function _createAuth($auth, $options)
         return new $class($this, $options);
     }
 }
+// @deprecated Backwards compatibility with earler 3.x versions.
+class_alias('Cake\Http\Client', 'Cake\Network\Http\Client');
```

Comment:
I agree that userland code shouldn't be defining classes in a namespace they don't own, and that is a reasonable reply to anyone who hits this exotic condition.